python/cudf/cudf/core/index.py

```diff
@@ -2122,6 +2204,21 @@ def _get_dt_field(self, field):
     def is_boolean(self):
         return False
 
+    def _clean_nulls_from_index(self):
```

Comment:
AFAICT, we've defined this function a total of five times now, across different classes in the `Index` hierarchy with similar docstrings.

* Can we write a single docstring for the base class implementatioin `BaseIndex._clean_nulls_from_index` - it will be "inherited" by all derived classes? If we need to document some behavior specific to the derived class implementation, we can use regular `# comments` instead.
* Is it not possible to reuse the base class implementation with`super()` at all here?